R/force_sorted.R

```diff
@@ -0,0 +1,71 @@
+#' @title Forcing a numeric vector into a monotonously increasing sequence.
+#'
+#' @description
+#' This function performs interpolation on the non-increasing parts of an
+#' input vector to ensure its values are monotonously increasing.
+#' If the values are non-increasing at the end of the vector, these values
+#' will be monotonously increasing by the value defined with parameter `by `
+#' from the last increasing value.
+#'
+#' @param x `numeric` vector.
+#'
+#' @param by `numeric(1)` value that will determine the monotonous increase in
+#'        case the values at the end of the vector are non-increasing and
+#'        therefore interpolation would not be possible. Default
+#'        to `by = .Machine$double.eps` which is the smallest positive
+#'        floating-point number x such that 1 + x != 1.
+#'
+#' @return A vector with continuously increasing values.
+#'
+#' @note
+#' NA values will not be replaced.
```

Comment:
Maybe also be a bit more clear here: *NA values in `x` will not be replaced and returned as is.*.